client/signup/steps/site-style/index.jsx

```diff
@@ -70,6 +70,64 @@ export class SiteStyleStep extends Component {
+					/>
+				</g>
+			</svg>
+		);
+	}
+
+	renderCalmSVG() {
+		return (
+			<svg width="80" height="32" xmlns="http://www.w3.org/2000/svg">
+				<g transform="translate(1 8)" fill="none" fill-rule="evenodd">
+					<path
+						d="M14.939 7.664c0-3.78 2.628-6.444 6.354-6.444 3.006 0 5.274 1.692 5.994 4.428h-3.384c-.522-1.08-1.476-1.62-2.646-1.62-1.89 0-3.186 1.404-3.186 3.636s1.296 3.636 3.186 3.636c1.17 0 2.124-.54 2.646-1.62h3.384c-.72 2.736-2.988 4.41-5.994 4.41-3.726 0-6.354-2.646-6.354-6.426zM28.565 8.96c0-3.168 1.962-5.148 4.428-5.148 1.494 0 2.556.684 3.096 1.566V3.956h3.078V14h-3.078v-1.422c-.558.882-1.62 1.566-3.114 1.566-2.448 0-4.41-2.016-4.41-5.184zm7.524.018c0-1.602-1.026-2.484-2.196-2.484-1.152 0-2.196.864-2.196 2.466 0 1.602 1.044 2.502 2.196 2.502 1.17 0 2.196-.882 2.196-2.484zM41.399 14V.68h3.078V14h-3.078zm19.152 0V8.546c0-1.314-.756-2.034-1.926-2.034s-1.926.72-1.926 2.034V14h-3.06V8.546c0-1.314-.756-2.034-1.926-2.034s-1.926.72-1.926 2.034V14h-3.078V3.956h3.078v1.26c.594-.81 1.62-1.368 2.934-1.368 1.512 0 2.736.666 3.42 1.872a4.04 4.04 0 0 1 3.384-1.872c2.484 0 4.086 1.584 4.086 4.284V14h-3.06z"
+						fill="#242424"
+					/>
+					<rect stroke="#232A2E" stroke-width="2" y="4" width="8" height="8" rx="2" />
+					<rect stroke="#232A2E" stroke-width="2" x="70" y="4" width="8" height="8" rx="2" />
+				</g>
+			</svg>
+		);
+	}
+
+	renderModernSVG() {
+		return (
+			<svg width="62" height="32" xmlns="http://www.w3.org/2000/svg">
+				<g fill="none" fill-rule="evenodd">
+					<path
+						d="M9.805 26v-8.244h-.054l-.918 1.836-2.448 4.446-2.394-4.428-.954-1.98h-.054V26H.409V13.436h2.898l3.078 5.886h.036l3.042-5.886h2.916V26H9.805zm9.036.216c-2.772 0-4.428-1.908-4.428-4.95 0-3.042 1.656-4.932 4.428-4.932 2.772 0 4.428 1.89 4.428 4.932 0 3.042-1.656 4.95-4.428 4.95zm0-2.106c1.044 0 1.656-.684 1.656-1.872v-1.926c0-1.188-.612-1.872-1.656-1.872s-1.656.684-1.656 1.872v1.926c0 1.188.612 1.872 1.656 1.872zM30.955 26v-1.584h-.09c-.27 1.026-1.314 1.8-2.538 1.8-2.394 0-3.636-1.746-3.636-4.95s1.242-4.932 3.636-4.932c1.224 0 2.268.756 2.538 1.8h.09V12.68h2.664V26h-2.664zm-1.746-1.944c1.008 0 1.746-.522 1.746-1.386v-2.79c0-.864-.738-1.386-1.746-1.386s-1.746.774-1.746 1.962v1.638c0 1.188.738 1.962 1.746 1.962zm10.836 2.16c-2.898 0-4.536-1.926-4.536-4.95 0-2.988 1.584-4.932 4.428-4.932 3.132 0 4.392 2.268 4.392 4.824v.792h-6.066v.144c0 1.242.666 2.034 2.088 2.034 1.116 0 1.728-.522 2.322-1.134l1.332 1.656c-.846.972-2.232 1.566-3.96 1.566zm-.054-7.92c-1.062 0-1.728.774-1.728 1.944v.144h3.312v-.162c0-1.152-.522-1.926-1.584-1.926zM48.883 26h-2.664v-9.45h2.664v2.052h.09c.216-1.044.936-2.052 2.412-2.052h.468v2.484h-.666c-1.548 0-2.304.342-2.304 1.44V26zm7.272 0h-2.664v-9.45h2.664v1.602h.108c.342-.99 1.134-1.818 2.628-1.818 1.926 0 2.952 1.368 2.952 3.744V26h-2.664v-5.706c0-1.206-.342-1.8-1.314-1.8-.864 0-1.71.45-1.71 1.386V26z"
+						fill="#181818"
+					/>
+					<path fill="#080808" d="M7 7h47v3H7z" />
+				</g>
+			</svg>
+		);
+	}
+
```

Comment:
Something about including all this svg code in the component bugs me, though if I were pressed to elaborate, I'd only come up with a muddling statement about cluttering the IDE window.

But I went ahead and tested using straight image tags anyway, e.g.,

```{html}
<img className="site-style__button-image" src={ `/calypso/images/signup/site-style-button-${ siteStyleProperties.label.toLowerCase() }.svg` } alt={ siteStyleProperties.label } />
```
There was a slight, but discernible 'loading' effect when refreshing an uncached page.

So I tried whacking them in a background image to reduce the effect:

```{html}
<span className={ `site-style__button-image is-${ siteStyleProperties.label.toLowerCase() }` } title={ siteStyleProperties.label } />
```

```{scss}
	.site-style__button-image {
		...
		background-repeat: no-repeat;
		background-position: top center;
		background-image: url( '/calypso/images/signup/site-style-button-professional.svg' );
		&.is-calm {
			background-image: url( '/calypso/images/signup/site-style-button-calm.svg' );
		}
		&.is-sophisticated {
			background-image: url( '/calypso/images/signup/site-style-button-sophisticated.svg' );
		}
		&.is-modern {
			background-image: url( '/calypso/images/signup/site-style-button-modern.svg' );
		}
	}
```

This was better, but may not be perfect. If it's all too much trouble, let's just leave it for now.

Reply:
OR :)

Throw them all into an external file, e.g., `site-style-buttons.jx`

```
// USAGE: <SiteStyleButton id={ siteStyleProperties.id }/>
const SiteStyleButton = ( { id } ) => {
	const buttons = {
		sophisticated: <svg width="93" height="32" xmlns="http://www.w3.org/2000/svg">
          ....
		</svg>,
	};

	return buttons[ id ];
};

export default SiteStyleButton;

```

Reply:
I like this option. It's cleaner and avoids the loading issue we see with the image tag or css. Thanks for the suggestion!

Reply:
@ramonjd: could we not add them to the `siteStyleOptions` object in site-style.js?

Reply:
@michaeldcain 

> could we not add them to the `siteStyleOptions` object in site-style.js? It's essentially an API response, and we have a pattern for doing similar with the icons on the segments endpoint for mobile.

It's the logical place for it, agree. The reason I left it out was that I thought it was intended to be an API response... to keep UI and data separate, but if there's a precedent elsewhere (**TIL**), let's do it. 👍

Reply:
> to keep UI and data separate

Yeah - I think this could be a grey area in that regard. We return icon urls for segments for the mobile app, which are technically data _for_ the UI. With the design pattern of components that support icons (i.e. [accordian menus](https://wpcalypso.wordpress.com/devdocs/design/accordion) or the [select dropdown](https://wpcalypso.wordpress.com/devdocs/design/select-dropdown)), it feels like this is the type of information that could be returned in the API response.